src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java

```diff
@@ -23,64 +26,35 @@ public class CuB2BDaoImpl extends B2BDaoImpl {
-				charSet = "ISO-8859-1";
-				LOG.info("content-type is multipart/related; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
-			} else {
-				conn.setRequestProperty("Content-type", "text/xml");
-
-				LOG.info("content-type is text/xml");
-			}
-
-            OutputStream out = conn.getOutputStream();
-            OutputStreamWriter outw = new OutputStreamWriter(out, charSet);
-            outw.write(request);
-            outw.flush();
-            outw.close();
-            out.flush();
-            out.close();
-
-            InputStream inp = conn.getInputStream();
-
-            StringBuffer response = new StringBuffer();
-            int i = inp.read();
-            while (i >= 0) {
-                if (i >= 0) {
-                    response.append((char) i);
-                }
-                i = inp.read();
-            }
-            return response.toString();
-        }
-        catch (MalformedURLException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (ProtocolException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (UnsupportedEncodingException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (IOException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
+    public String sendPunchOutRequest(final String request, final String punchoutUrl) {
+    	 LOG.debug("sendPunchOutRequest() started");
+
+         try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+             final HttpPost httpPost = new HttpPost(punchoutUrl);
+             if (request.contains("MIME_BOUNDARY_FOR_ATTACHMENTS")) {
+            	// KFSPTS-794 : for attachments
+                 httpPost.addHeader(HttpHeaders.ACCEPT, MediaType.MULTIPART_RELATED);
+                 httpPost.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.MULTIPART_RELATED + "; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
+                 httpPost.setEntity(new StringEntity(request, StandardCharsets.ISO_8859_1));
+                 LOG.info("content-type is multipart/related; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
+             }
+             else {
+                 httpPost.addHeader(HttpHeaders.ACCEPT, MediaType.TEXT_XML);
+                 httpPost.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_XML);
+                 httpPost.setEntity(new StringEntity(request));
```

Comment:
The old customization treats the request content as UTF-8 if no attachments are present, though it might not have changed the request's content type settings. Should this line be updated to set that charset on the request, or should we leave it as ISO-8859-1 instead?

Reply:
yes, I think this would make sense. Adding.